### PR TITLE
Move test database setup to separate module

### PR DESCRIPTION
**Pull request by Ponewor**
Requires: #67

### CODE REVIEW

File: scylla-cdc-printer/src/printer.rs [hunk 1]
```diff
@@ -46,70 +42,27 @@ mod tests {
...
         populate_db_with_pk(&shared_session, partition_key_2)
             .await
             .unwrap();
 
+        let start = chrono::Duration::from_std(
+            SystemTime::now()
+                .duration_since(SystemTime::UNIX_EPOCH)
+                .unwrap(),
+        )
+            .unwrap();
```
Comment (piodul) [inline scylla-cdc-printer/src/printer.rs]: Let's schedule #66 for merge first so that you can use `chrono::Local::now().to_timestamp()` here.

Apart from that, there is an unnecessary indent before `.unwrap()`.

File: scylla-cdc/src/test_utilities.rs [hunk 2]
```diff
@@ -62,11 +67,23 @@ fn get_uri() -> String {
...
     let shared_session = Arc::new(session);
 
-    let ks = create_test_db(&shared_session).await?;
+    let ks = create_test_db(&shared_session, schema, replication_factor).await?;
     Ok((shared_session, ks))
-}
\ No newline at end of file
+}
+
+pub async fn prepare_simple_db() -> anyhow::Result<(Arc<Session>, String)> {
```
Comment (piodul) [inline scylla-cdc/src/test_utilities.rs:82]: I believe the commit diffs will look a bit nicer if you introduce and start using this function in the previous commit. You won't have to do so many `db` -> `simple_db` renames.

Besides, it looks like `prepare_simple_db`'s body could be simplified to just:

```rust
{
    prepare_db(&[get_create_table_query()], 1).await
}
```

File: scylla-cdc/src/test_utilities.rs [hunk 3]
```diff
@@ -25,25 +25,30 @@ fn get_create_table_query() -> String {
...
     session.use_keyspace(&ks, false).await?;
 
-    // Create test table
-    let create_table_query = get_create_table_query();
-    session.query(create_table_query, &[]).await?;
-    session.await_schema_agreement().await?;
+    // Create test tables
+    for query in schema {
+        session.query(query.clone(), &[]).await?;
+    }
```
Comment (piodul) [inline scylla-cdc/src/test_utilities.rs:46]: You accidentally removed the schema agreement wait.

File: scylla-cdc-replicator/src/replication_tests.rs [hunk 4]
```diff
@@ -522,9 +473,7 @@ mod tests {
             ),
         ];
 
-        test_replication(&get_uri(), schema, operations)
-            .await
-            .unwrap();
+        test_replication(schema, operations).await.unwrap();
```
Comment (piodul) [inline scylla-cdc-replicator/src/replication_tests.rs:476]: Getting rid of the `node_uri` parameter sounds like a change that should be put in a separate commit.

**Review by piodul — CHANGES_REQUESTED** [review]
That's a nice improvement. I left some comments. Apart from them, please change the title of the PR to something more descriptive - "Refactor tests" doesn't tell much about how the tests were refactored.

File: scylla-cdc-replicator/src/replication_tests.rs [hunk 5]
```diff
@@ -969,8 +890,6 @@ mod tests {
             "DELETE FROM RANGE_DELETE WHERE pk1 = 0 AND pk2 = 0 AND (ck1, ck2, ck3) > (3, 3, 3)",
         ]);
 
-        test_replication(&get_uri(), schema, operations)
-            .await
-            .unwrap();
+        test_replication(schema, operations).await.unwrap();
```
Comment (piodul) [inline scylla-cdc-replicator/src/replication_tests.rs:893]: The last-but-one commit in the PR does not compile. Probably because this change belongs to the last-but-one commit, not the last one.

Comment (Ponewor) [reply scylla-cdc-replicator/src/replication_tests.rs:893]: This one is a rebasing hell at this point 😩

**Comment by Ponewor** [comment]
@piodul what's blocking us here?

**Review by piodul — APPROVED** [review]
